### PR TITLE
add swagger base

### PR DESCRIPTION
**Pull request by justincorrigible**
Note: This PR only sets up the Swagger ui, and exposes its endpoint.
It does not contain the actual documentation for the multiple API's endpoints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,10 +31,19 @@ Now you should be able to start the server from the project's root folder:
...
-# or before running the server
+  # or do this, before running the server
   export NODE_OPTIONS=--openssl-legacy-provider
 ```
+
+The development server will start on port `4000` by default.
+
+#### - Documentation
+
+There is API documentation provided by Swagger, available at [/swagger](http://localhost:4000/swagger).
```
Comment (justincorrigible) [inline README.md:49]: `Readme` improvements

File: package.json [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+  "description": "Data Access Control API",
   "main": "src/index.ts",
   "scripts": {
     "build": "ttsc",
-    "start": "TS_NODE_BASEURL=./dist node -r tsconfig-paths/register ./dist/index.js",
     "dev": "ts-node ./src/index.ts",
-    "dev:nodemon": "nodemon -w src -e ts,json -x ts-node ./src/index.ts",
+    "dev:nodemon": "nodemon -e js,json,ts -x ts-node ./src/index.ts",
+    "start": "TS_NODE_BASEURL=./dist node -r tsconfig-paths/register ./dist/index.js",
+    "swagger": "swagger-jsdoc -d ./dist/swagger.js -o ./dist/swagger/documentation.json",
```
Comment (justincorrigible) [inline package.json:11]: add `swagger` npm script, and improve `nodemon`'s to restart the API when working on configs and/or with linked  dependencies (such as Arranger 😉)

File: src/constants/endpoint.ts [hunk 3]
```diff
@@ -1,2 +1,3 @@
 export const BASE_ENDPOINT = '/api';
 export const HEALTH_ENDPOINT = '/health';
+export const SWAGGER_ENDPOINT = '/swagger';
```
Comment (justincorrigible) [inline src/constants/endpoint.ts:3]: new endpoint constant, used both at root, and inside of `/api` to redirect `api/docs` as a gesture of decency 🧐

File: src/routes/index.ts [hunk 4]
```diff
@@ -11,9 +13,14 @@ const routes = (): Router => {
   router.use('/download/sequences', sequenceDownloadRouter);
   router.use('/storage', storageRouter);
 
+  router.get('/docs', (req, res) => {
+    res.status(302).redirect(SWAGGER_ENDPOINT);
```
Comment (justincorrigible) [inline src/routes/index.ts:17]: redirection for great good

File: src/routes/storage/storage.router.ts [hunk 5]
```diff
@@ -48,6 +48,15 @@ storageRoute.all(async (req: Request, res: Response, next: NextFunction) => {
...
 
+/**
+ * @swagger
+ * /storage:
+ *  get:
+ *    responses:
+ *      200:
+ *        description: Returns all the contents stored by the user.
+ *    tags: [storage]
+ */
```
Comment (justincorrigible) [inline src/routes/storage/storage.router.ts:59]: documentation placeholders to serve as (bad) example

File: src/routes/swagger/config.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import path from 'path';
+
+import manifest from '@/../package.json';
+
+const swaggerJsdocConfig = {
```
Comment (justincorrigible) [inline src/routes/swagger/config.ts:5]: the settings for the Swagger instance

File: src/routes/swagger/config.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      version: manifest.version,
+      description: manifest.description,
+    },
+  },
+  apis: [
+    path.resolve(
+      process.cwd(),
+      process.env.NODE_ENV === 'production'
+        ? 'dist/routes/**/*.router.js'
+        : 'src/routes/**/*.router.ts',
```
Comment (justincorrigible) [inline src/routes/swagger/config.ts:19]: I haven't tested this, but should enable swagger while deployed?

File: src/routes/swagger/swagger.router.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import swaggerJsdoc from 'swagger-jsdoc';
+import swaggerUi from 'swagger-ui-express';
+
+import config from './config';
+
+const router: Router = Router();
+const specs = swaggerJsdoc(config);
+
+router.use('/', swaggerUi.serve, swaggerUi.setup(specs));
```
Comment (justincorrigible) [inline src/routes/swagger/swagger.router.ts:10]: the actual endpoint

File: src/server.ts [hunk 9]
```diff
@@ -49,6 +49,7 @@ if (process.env.NODE_ENV === 'production') {
 
 app.use(BASE_ENDPOINT, apiRoutes());
 app.use(HEALTH_ENDPOINT, healthRouter);
+app.use(SWAGGER_ENDPOINT, swaggerRouter);
```
Comment (justincorrigible) [inline src/server.ts:52]: ... 👍

**Review by justincorrigible — COMMENTED** [review]
the usual...

**Review by joneubank — APPROVED** [review]

**Pull request merged**